pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -46,8 +46,8 @@ def __init__(
     ) -> None:
         self._model: Optional[Module] = None
         checkpoint_io = checkpoint_io if checkpoint_io is not None else TorchCheckpointIO()
-        self._checkpoint_io = checkpoint_io
-        self._precision_plugin = precision_plugin if precision_plugin is not None else PrecisionPlugin()
```

Comment:
The setters weren't getting used.

Reply:
And how are they getting used now? There can't be a public attribute and a setter with the same name.

Reply:
This will use the setter to set `self._precision_plugin`.

If we don't do it like this, any custom logic in the setter won't run